authors/apps/authentication/views.py

```diff
@@ -202,3 +214,99 @@ def patch(self, request, token):
             })
         except exc.BadSignature:
             raise exceptions.ValidationError("This is an invalidated link")
+
+
+class FacebookAPIView(APIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = FacebookAndGoogleSerializer
+
+    @swagger_auto_schema(
+        operation_description='Social Auth with Facebook',
+        operation_id='Login in a user using their Facebook credentials',
+        request_body=serializer_class,
+        responses={200: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def post(self, request):
+
+        user_data = request.data.get("user", {})
+        get_token = user_data.get("access_token")
+        # get the token
+        try:
+            facebook_acct_user = facebook.GraphAPI(access_token=get_token)
+            user_details = facebook_acct_user.get_object(
+                id='me', fields='id, name, email')
+
+            facebook_user = check_user.validate_system_user(user_details)
+
+            return Response(facebook_user, status=status.HTTP_200_OK)
+            
+        except:
+            return Response(
+                {"error": "Facebook login failed. Token is expired or invalid"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class GoogleAPIView(APIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = FacebookAndGoogleSerializer
+
+    @swagger_auto_schema(
+        operation_description='Social Auth with Google',
+        operation_id='Login in a user using their google credentials',
+        request_body=serializer_class,
+        responses={200: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def post(self, request):
+
+        user_data = request.data.get("user", {})
+        googl_auth_token = user_data.get("access_token")
+        # get the token
+        try:
+            user_cred = id_token.verify_oauth2_token(
+                googl_auth_token, requests.Request())
+
+            verified_user = check_user.validate_system_user(user_cred)
+
+            return Response(verified_user, status=status.HTTP_200_OK)
+            
+        except:
+            return Response(
+                {"error": "google login failed. Token is either invalid or expired"}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This line is too long, I think it violates the standards.

Reply:
I thought so too, but autopep8 confirmed it does not. Try running `autopep8 --in-place --aggressive --aggressive` and confirm.